shap/explainers/_tree.py

```diff
@@ -1648,6 +1661,18 @@ def _recalculate_value(tree, i , level):
             self.features = np.where(self.features >= 0, tree_features[self.features], self.features)
 
 
+def get_dmatrix_properties(model):
+    """
+    Retrieves properties from an xgboost.sklearn.XGBModel instance that should be passed to the xgboost.core.DMatrix object before calling predict on the model
+    """
+    properties_to_pass = ["missing", "n_jobs", "enable_categorical", "feature_types"]
+    dmatrix_attributes = {}
+    for attribute in properties_to_pass:
+        value = getattr(model, attribute)
+        if value is not None:
+            dmatrix_attributes[attribute] = value
```

Comment:
I'm not an xgboost maintainer so can't say with certainty but I don't think there would be an issue if they were none. In fact, I originally included them all even if None but @CloseChoice asked me to add the if check. IMO there is no need to pollute the kwargs with Nones since those are the defaults in most cases in the xgboost library so could make sense to leave out. The only issue is if a user explicitly sets a setting to None and we omit it then there could be some unwanted behaviour so maybe best to keep?

Reply:
At present, `getattr(model, attribute)` will raise an `AttributeError` if the given attribute is not found, which seems undesirable. It would be better to pass on only those attributes which are present.

Also, as you say, we should probably allow users to define `None` as a possible value. So, we could avoid using `None` as a missing value sentinel.

How about something like:

```python
if hasattr(model, attribute):
    dmatrix_attributes[attribute] = getattr(model, attribute)
```

Reply:
Good spot. I agree its better practice to safeguard this. All `xgboost.sklearn.XGBModel` instances initialise the above attributes with sane defaults either way but in case their API changes in the future this is a more robust way of doing it. I have amended with your suggestion :)

Reply:
yep makes sense, thanks